File: iosApp/iosApp/Modules/DS/Components/Photo/PhotoDate.swift
//
//  PhotoDate.swift
//  PicProgress
//
//  Created by Jeferson on 30/01/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import SwiftUI
import shared

struct PhotoDate: View {
    var album: Album
    var photo: Photo

    var body: some View {
        VStack(spacing: 4) {
            Text(getLabelText())
                .font(.titleSmall())
                .frame(maxWidth: .infinity)
            Text(photo.createdAt.date.format())
                .font(.bodySmall())
                .frame(maxWidth: .infinity)
        }
        .padding(16)
    }

    func getLabelText() -> String {
        switch album.frequency {
        case .daily:
            return String(format: NSLocalizedString("Day", comment: ""), photo.createdAt.date.getDayNumber(referenceDate: album.createdAt.date))
        case .weekly:
            return String(format: NSLocalizedString("Week", comment: ""), photo.createdAt.date.getWeekNumber(referenceDate: album.createdAt.date))
        case .monthly:
            return String(format: NSLocalizedString("Month", comment: ""), photo.createdAt.date.getMonthNumber(referenceDate: album.createdAt.date))
        default:
            return ""
        }
    }
}

File: androidApp/src/main/java/com/picprogress/ds/components/photo/PhotoDate.kt
package com.picprogress.ds.components.photo

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.picprogress.R
import com.picprogress.model.album.Album
import com.picprogress.model.album.Frequency
import com.picprogress.model.photo.Photo
import com.picprogress.util.format
import com.picprogress.util.getDayNumber
import com.picprogress.util.getMonthNumber
import com.picprogress.util.getWeekNumber

@Composable
fun PhotoDate(album: Album, photo: Photo) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        Text(
            text = when (album.frequency) {
                Frequency.DAILY -> stringResource(id = R.string.photo_day_label, photo.createdAt.date.getDayNumber(album.createdAt.date))
                Frequency.WEEKLY -> stringResource(id = R.string.photo_week_label, photo.createdAt.date.getWeekNumber(album.createdAt.date))
                Frequency.MONTHLY -> stringResource(id = R.string.photo_month_label, photo.createdAt.date.getMonthNumber(album.createdAt.date))
            },
            textAlign = TextAlign.Center,
            modifier = Modifier.fillMaxWidth(),
            style = MaterialTheme.typography.titleSmall
        )
        Text(
            text = photo.createdAt.date.format(),
            textAlign = TextAlign.Center,
            modifier = Modifier.fillMaxWidth(),
            style = MaterialTheme.typography.bodySmall
        )
    }
}